[BUG] Typescript error: The type for webServer config property (TestConfigWebServer) is not typed correctly

**Context:**
- Playwright Version: [1.24.0]
- Operating System: [WSL2 (Ubuntu)]
- Node.js version: [v16.13.0]
- Browser: [All]

<!-- CLI to auto-capture this info -->
<!-- npx envinfo --preset playwright --markdown -->

**Code Snippet**
Here is the config that I use where the typescript error occurs
```TypeScript
import { devices } from "@playwright/test";
import type { PlaywrightTestConfig } from "@playwright/test";

const launchScript = process.env.CI ? "start" : "dev";
const testWebPort = 3000;

const config: PlaywrightTestConfig = {
  testMatch: /\/tests\/.+\.test\.ts$/,
  forbidOnly: !!process.env.CI,
  retries: process.env.CI ? 2 : 0,
  reporter: process.env.CI ? "github" : "list",
  webServer: [
    {
      command: `yarn workspace @syncbase/web next ${launchScript}`,
      env: { NEXT_PUBLIC_IS_TEST: "1", PORT: testWebPort },
      port: testWebPort,
    },
    {
      command: `yarn workspace @syncbase/server ${launchScript}`,
      env: { NODE_ENV: "test" },
      port: 8082,
    },
  ],
  use: {
    baseURL: `http://localhost:${testWebPort}`,
    trace: "on-first-retry",
  },
  workers: 1,
  projects: [
    {
      name: "Desktop Chrome",
      use: { ...devices["Desktop Chrome"] },
    },
    {
      name: "Desktop Firefox",
      use: { ...devices["Desktop Firefox"] },
    },
    {
      name: "Desktop Safari",
      use: { ...devices["Desktop Safari"] },
    },
  ],
};

export default config;

```

**Describe the bug**
As you can see here is the typescript error:

<img width="587" alt="image" src="https://user-images.githubusercontent.com/28491857/180618681-1bc30d11-bd4c-4a6c-b673-0650c488db47.png">

When I call `tsc --noEmit` this is what I get:
```
tests/playwright.config.ts:12:3 - error TS2741: Property 'command' is missing in type '({ command: string; env: { NEXT_PUBLIC_IS_TEST: string; PORT: number; }; port: number; } | { command: string; env: { NODE_ENV: string; }; port: number; })[]' but required in type 'TestConfigWebServer'.

12   webServer: [
     ~~~~~~~~~

  ../../node_modules/@playwright/test/types/test.d.ts:4446:3
    4446   command: string;
           ~~~~~~~
    'command' is declared here.
  ../../node_modules/@playwright/test/types/test.d.ts:490:3
    490   webServer?: TestConfigWebServer;
          ~~~~~~~~~
    The expected type comes from property 'webServer' which is declared here on type 'PlaywrightTestConfig<{}, {}>'


Found 1 error in tests/playwright.config.ts:12

```

--- a/packages/playwright-test/types/test.d.ts
+++ b/packages/playwright-test/types/test.d.ts
@@ -487,7 +487,7 @@ interface TestConfig {
    * ```
    *
    */
-  webServer?: TestConfigWebServer;
+  webServer?: TestConfigWebServer | TestConfigWebServer[];
   /**
    * Configuration for the `expect` assertion library. Learn more about [various timeouts](https://playwright.dev/docs/test-timeouts).
    *

--- a/tests/playwright-test/config.spec.ts
+++ b/tests/playwright-test/config.spec.ts
@@ -391,3 +391,29 @@ test('should work with undefined values and base', async ({ runInlineTest }) => 
   expect(result.exitCode).toBe(0);
   expect(result.passed).toBe(1);
 });
+
+test('should have correct types for the config', async ({ runTSC }) => {
+  const result = await runTSC({
+    'playwright.config.ts': `
+      import type { PlaywrightTestConfig } from '@playwright/test';
+
+      const config: PlaywrightTestConfig = {
+        webServer: [
+          {
+            command: 'echo 123',
+            env: { PORT: '123' },
+            port: 123,
+          },
+          {
+            command: 'echo 123',
+            env: { NODE_ENV: 'test' },
+            port: 8082,
+          },
+        ],
+      };
+      
+      export default config;
+  `
+  });
+  expect(result.exitCode).toBe(0);
+});

--- a/utils/generate_types/overrides-test.d.ts
+++ b/utils/generate_types/overrides-test.d.ts
@@ -58,7 +58,7 @@ type LiteralUnion<T extends U, U = string> = T | (U & { zz_IGNORE_ME?: never });
 
 interface TestConfig {
   reporter?: LiteralUnion<'list'|'dot'|'line'|'github'|'json'|'junit'|'null'|'html', string> | ReporterDescription[];
-  webServer?: TestConfigWebServer;
+  webServer?: TestConfigWebServer | TestConfigWebServer[];
 }
 
 export interface Config<TestArgs = {}, WorkerArgs = {}> extends TestConfig {

